src/main/java/com/splunk/logging/HttpEventCollectorLogbackAppender.java

```diff
@@ -97,21 +99,52 @@ public void stop() {
     }
 
     @Override
-    protected void append(ILoggingEvent event) {
+    protected void append(E e) {
+        if (e instanceof ILoggingEvent) {
+            sendEvent((ILoggingEvent) e);
+        } else {
+            sendEvent(e);
+        }
+    }
+
+    private void sendEvent(ILoggingEvent event) {
         event.prepareForDeferredProcessing();
-        event.getCallerData();
+        if (event.hasCallerData()) {
+            event.getCallerData();
+        }
+
         MarkerConverter c = new MarkerConverter();
         if (event != null && started) {
             this.sender.send(
                     event.getLevel().toString(),
-                    _layout.doLayout(event),
+                    _layout.doLayout((E) event),
                     _includeLoggerName ? event.getLoggerName() : null,
                     _includeThreadName ? event.getThreadName() : null,
                     _includeMDC ? event.getMDCPropertyMap() : null,
                     (!_includeException || event.getThrowableProxy() == null) ? null : event.getThrowableProxy().getMessage(),
                     c.convert(event)
-                    );
+            );
+        }
+    }
+
+    // send non ILoggingEvent such as ch.qos.logback.access.spi.IAccessEvent
+    private void sendEvent(E e) {
+        String message = _layout.doLayout(e);
+        if (message == null) {
+            throw new IllegalArgumentException(String.format(
+                    "The logback layout %s is probably incorrect, " +
+                            "and fails to format the message.",
+                    _layout.toString()));
         }
+        this.sender.send(
```

Comment:
OK, let's make a helper method in `HttpEventCollectorSender`. Something like this:

```java
public void send(String message) {
    send("", message, "", "", null, null, "");
}
```